.github/workflows/tsc.yml

```
@@ -0,0 +1,18 @@
+name: Type Check
+
+on: push
+
+jobs:
+   tsc:
+      runs-on: ubuntu-latest
+      steps:
+         - name: Checkout
+           uses: actions/checkout@v2
+         - name: Install PNPM
+           run: corepack enable && corepack prepare pnpm@6.32.11 --activate
+         - name: Install workspaces
+           env:
+              FONT_AWESOME_NPM_TOKEN: ${{ secrets.FONT_AWESOME_NPM_TOKEN }}
+           run: pnpm install:all
+         - name: Run Typescript Compiler
+           run: bin/type-check.sh
```

I think we can just use the npm script, which is also the one that we run on git hook.

```suggestion
           pnpm type-check
```

You can extend it to check the backend folder using the same approach used for the dev script (`npm-run-all`)

---

Good idea, thanks for the suggestion.

I noticed that `pnpm type-check:frontend` overlooked the type errors in `@ifixit/ui` that are being fixed here https://github.com/iFixit/react-commerce/pull/738. Other times, it catches the type errors in `packages/`. Perhaps it's only checking code that is being imported from `packages/` to `frontend/`?

This motivated me to explicitly run `tsc` on all packages to always catch type errors (see https://github.com/iFixit/react-commerce/pull/733/commits/12374962062d76059e34ad191cedf30f420a5f8c). However, this can result in running `tsc` on the same code more than once, which slows things down. What are your thoughts on this?

---

@masonmcelvain Yeah I was actually surprised that type checking the frontend wasn't type checking also the dependent packages. I think your approach of type-checking every package is the way to go, and I've found a "more native" pnpm way to do it using this command:

```sh
pnpm run -r type-check
```

The `-r` flag tells pnpm to run the `type-check` script of every app or package defined in `pnpm-workspace.yaml`.
So it's sufficient to declare the `type-check` script on every package, we don't need to define the glue scripts ourselves.
Notably `backend` is not specified as part of the monorepo (due to issues installing Strapi dependencies as part of the monorepo), and that's fine we can just write an additional script for that